### ISSUE TITLE
EXC_BAD_ACCESS when using VC that extends UIViewController with generics.

### ISSUE THREAD

**Issue by llinardos** [johnno1962/InjectionIII#57]
Hi guys!

I'm having an issue when trying to inject a VC that extends a UIViewController with generics.

I have:
```
class GenericVC<V: UIView> : UIViewController {
  lazy var mainView: V = { return V.init() }()
  override func loadView() {
    super.loadView()
    self.view = mainView
  }
}
```

I use it like:

```
class DummyGenericVC: GenericVC<DummyView> {}
```

and I present the VC as:

```
let vc = DummyGenericVC()
UIApplication.shared.keyWindow?.rootViewController = vc
```

When is injected I got:

```
Thread 1: EXC_BAD_ACCESS (code=1, address=0x0)
```

This not happen if I use `GenericVC<DummyView>` like:

```
let vc = GenericVC<DummyView>
UIApplication.shared.keyWindow?.rootViewController = vc
```

But is not a possibility for the way we are working in the project.

I'm attaching [a project](https://github.com/johnno1962/InjectionIII/files/2492872/GenericVCInjectionIssue.zip) showing the issue. When `let useGenericVC = true` the crash happens, when `let useGenericVC = false` injection works great as always.

I know is not a common case, but we use this approach extensively.

**Comment by johnno1962** [comment]
Hi, interesting problem. I’ll be honest I’m surprised generics are working at all!

One thing you can try is creating a factory class that does the object creation inside the source file with your generic classes that are being injected so the object creation is also injected:
```
class Factory {
    func presentVC(generic: Bool) {
        let vc = generic ? DummyGenericVC() : GenericVC<DummyView>()
        UIApplication.shared.keyWindow?.rootViewController = vc
    }
}
```
Create an instance of this class in your AppDelegate that isn’t being injected and message it:
````
factory.presentVC(generic: useGenericVC)
```
This no longer crashes.

**Comment by llinardos** [comment]
That's right man! Thanks!

I've detected that the sample project attached in the first message, works ok with iOS 12 simulator. But crashes with the same error in iOS 10.3.1 simulators.

Anyway, In the "real" project were we're using this extending-generic-VCs approach, I changed to iOS 12 simulator + factory approach (in the SAME FILE of the extending-generic-VC) and everything is working ok. 

Thanks again! This is an amazing dev-tool 💪
